### PR TITLE
Fix multi-threading crash in Poppler CMS rendering (fixes #144)

### LINKED ISSUE #144: Crash: double free or corruption (!prev)
Boomaga crashes frequently with the message "double free or corruption (!prev)"
Using Boomaga 3.0.0 on Ubuntu 25

### PR DESCRIPTION
**Pull request by hoky24**
Fixes #144.

Poppler builds its process-wide LittleCMS state lazily in the `GfxState`
constructor, guarded by nothing but a plain `if (!ptr)` on a static:
`GfxState::sRGBProfile`, `GfxXYZ2DisplayTransforms::XYZProfile` and the CMS
error handler. Boomaga starts 16 render workers at once, several of them enter
that code concurrently, and the heap is corrupted.

The fix builds that state once from a single thread first. The constructor runs
for every render regardless of page content, so one arbitrary page is enough, at
1 dpi since only the side effect matters. The guard is set only after a render
actually succeeded, so an unreadable first document does not disable the
protection for the ones loaded later.

Rendering a throwaway page is not elegant, but `poppler-cpp` exposes nothing for
colour management and those statics are private, reachable only through a
render. The proper fix belongs upstream and is reported there, with a tested
patch, as https://gitlab.freedesktop.org/poppler/poppler/-/work_items/1800 - the
comment in the code points at it, so it is clear when this can be dropped again.

Testing: a standalone 16-thread reproducer crashes in 16 of 200 runs on Poppler
26.09.0 and 19 of 200 on 26.06.0 without the warm-up, and 0 of 200 with it.
Boomaga builds clean with Qt6 and was tested on NixOS - started the normal way
by printing to it, then several further PDFs opened including a 300 page one,
no crash.

This covers the CMS globals only. It is not a claim that Poppler is thread safe
for concurrent rendering in general.

### CODE REVIEW

**Comment by archisman-panigrahi** [comment]
@hoky24 The comment https://github.com/Boomaga/boomaga/issues/144#issuecomment-5743182030 says the bug is absent, but maybe it is still present in some edge case. Can you please describe how I can reproduce the bug, so that I can test it before I merge this PR?

Thank you for sending the PR!

**Comment by hoky24** [comment]
> @hoky24 The comment [#144 (comment)](https://github.com/Boomaga/boomaga/issues/144#issuecomment-5743182030) says the bug is absent, but maybe it is still present in some edge case. Can you please describe how I can reproduce the bug, so that I can test it before I merge this PR?
> 
> Thank you for sending the PR!

It is a race, so "no crashes so far" is not evidence either way - and it reproduces on demand with the repository's own test file. Looping the unpatched build over `src/boomaga/tests/data/testInFiles/00-simple.pdf`:

```
for i in $(seq 1 100); do QT_QPA_PLATFORM=offscreen timeout 3 boomaga 00-simple.pdf; done
```

aborts with `double free or corruption (fasttop)` - the same message @katnatek reports in that thread - in 16 of 100 runs on a 24 core machine; the build from this PR, run interleaved under the same conditions, aborts in 0 of 100. The file does not matter (every PDF I tried does it at a few percent), the window is the first ~600 ms after start when the 24 workers begin rendering at once, and it will be rarer on a machine with few cores. Run it without gdb; the slowdown hides the race.

I have also pushed an update since your comment: Copilot's review showed the separate throwaway render was not needed, so the first real render now simply runs under a lock. The description is updated to match.

File: src/boomaga/render.cpp [hunk 1]
```diff
@@ -35,11 +37,48 @@
...
+        return;
+
+    std::unique_ptr<poppler::page> page(doc->create_page(0));
+    if (!page)
+        return;
+
+    // Only the side effect of constructing the GfxState matters, the image
+    // itself is thrown away, so render as few pixels as possible.
+    poppler::page_renderer().render_page(page.get(), 1.0, 1.0);
+    initialized = true;
```
[the remark this answers is not in the document]
Comment (hoky24) [reply src/boomaga/render.cpp]: Fixed in 7dc15f7, and it went a step further than the check suggested here. Tying the guard to the render succeeding would have made a document whose page 0 never renders repeat the throwaway render under the lock on every call, so the throwaway render is gone entirely: the first real render now runs under the lock and sets the guard when it succeeds.
